compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -2414,15 +2494,31 @@ private void validateIntersectionTypeDefinitions(List<BLangTypeDefinition> typeD
         }
     }
 
-    private void defineUndefinedReadOnlyTypes(List<BLangTypeDefinition> typeDefNodes, List<BLangNode> typDefs,
+    private void defineUndefinedReadOnlyTypes(List<BLangTypeDefinition> typeDefNodes, List<BLangNode> typeDefs,
                                               SymbolEnv pkgEnv) {
         // Any newly added typedefs are due to `T & readonly` typed fields. Once the fields are set for all
         // type-definitions we can revisit the newly added type-definitions and define the fields and members for them.
         populateImmutableTypeFieldsAndMembers(typeDefNodes, pkgEnv);
 
         // If all the fields of a structure are readonly or final, mark the structure type itself as readonly.
         // If the type is a `readonly object` validate if all fields are compatible.
-        validateFieldsAndSetReadOnlyType(typDefs, pkgEnv);
+        validateFieldsAndSetReadOnlyType(typeDefs, pkgEnv);
+
+        for (BLangNode typeDef : typeDefs) {
```

Comment:
This could be a method right, like `populateImmutableTypeFieldsAndMembers` and `validateFieldsAndSetReadOnlyType`.

Reply:
Fixed in 6b923b6.